curl/src/client.cc

```diff
@@ -133,14 +138,8 @@ namespace leatherman { namespace curl {
         curl_easy_reset(_handle);
 
         // Set common options
-        auto result = curl_easy_setopt(_handle, CURLOPT_NOPROGRESS, 1);
-        if (result != CURLE_OK) {
-            throw http_request_exception(req, curl_easy_strerror(result));
-        }
-        result = curl_easy_setopt(_handle, CURLOPT_FOLLOWLOCATION, 1);
-        if (result != CURLE_OK) {
-            throw http_request_exception(req, curl_easy_strerror(result));
-        }
+        curl_easy_setopt_maybe<long>(ctx, CURLOPT_NOPROGRESS, 1);
```

Comment:
The template parameter should be inferred from the arguments. No need to specify the `<long>`. Should be able to avoid defining all the callback signatures as well.